src/components/CreatureModal.tsx

```diff
@@ -0,0 +1,22 @@
+import './CreatureModal.css';
+
+type CreatureCardModalProps = {
+    image: string;
+    id: number;
+    name: string;
+    type: string;
```

Comment:
Since the type field has a fixed set of known values, it’s better to replace the plain string with a list of valid types, like: grass | water | fire | ....

Also, can a Pokémon have multiple types? If so, it would be better to use an array instead to support that structure.